src/service/mercury/queries.ts

```diff
@@ -0,0 +1,104 @@
+export const mutation = {
+  authenticate: `
+    mutation Auth {
+      authenticate(input: {email: $email, password: $password}) {
+        jwtToken
+      }
+    }
+  `,
+  newAccountSubscription: `
+    mutation MyMutation {
+      createFullAccountSubscription(
+        input: {fullAccountSubscription: {$pubKey: PUBKEY!, userId: 1}}
+      ) {
+        clientMutationId
+      }
+    }
+  `,
+};
+export const query = {
+  subscriptionById: `
+    query GetSubById {
+      contractEventById($id: ID!) {
+        id
+        data
+        contractId
+        ledgerTimestamp
+        nodeId
+        topic1
+        topic2
+        topic3
+        topic4
+      }
+    }
+  `,
+  allSubscriptions: `
+    query AllSubscriptions {
+      allContractEventSubscriptions {
+        edges {
+          node {
+            contractId
+          }
+        }
+      }
+    }
+  `,
+  getAccountHistory: `
+    query GetAccountHistory {
+      // need this to take array of contracts
+      query GetSubById {
+        eventByContractId($searchedContractId: CONTRACT_ID!) {
```

Comment:
Unless I'm missing something, this would require you to know in advance the ids of all the tokens the user holds. So you'd either have to store them in a store in this backend or retrieve them with a query (in a previous request, introducing overhead). I'd suggest structuring the fullaccount history by querying the events by topics so:
- eventByTopics [transfer, publicKey, _, _] (transfers from)
- eventByTopics [transfer, _, PublicKey, _] (transfers to)
 ... (same for mints, and approvals if you plan on indexing them)

This requires a longer query but removes the need for a store/pre-request. It also gives more context in the response about the structure of each result so the client doesn't have to check which kind of event every `[event {...}]` in the response is.

Reply:
Does `eventByTopics` take a contract_id? Not sure if I'm missing it but looking at the playground I only see that I can pass topics so would that match on any event with the matching topic, like ones from other unrelated contracts that happen to emit the same event?

Reply:
It would be pretty cool to be able to watch events by topic only though, maybe we could use that as a proxy for automatic token detection but that would probably have to come after something like this https://github.com/stellar/stellar-protocol/pull/1409 gained more adoption to avoid spam, etc.

Reply:
@aristidesstaffieri in Mercury every user has access to events that match their subscriptions, so you won't end up getting events from contracts you haven't subscribed to.

Reply:
ah yeah I see, thanks for the info. I'll try out the `eventByTopics` queries.